classes/task/merge_user_accounts.php

```diff
@@ -0,0 +1,218 @@
+            if ($retries >= $maxattempts) {
+                $status = merge_request::COMPLETED_WITH_ERRORS;
+                // Send a notification to administrator ?
+            } else {
+                $status = $statusforerror;
+            }
+        } else {
+            $status = merge_request::COMPLETED_WITH_SUCCESS;
+        }
+        $mergerequest->append_log($log, time());
+        if ($mergerequest->retries > $maxattempts) {
+            $log_attempts = "Reached the number of maximum attempts";
+            $mergerequest->append_log($log_attempts, time());
+            $status = merge_request::COMPLETED_WITH_ERRORS;
+        } 
+        $this->update_status_and_log($mergerequest->id,
+                                        $status,
+                                        $mergerequest->log);
+        $mergerequest->status = $status;
+        return $mergerequest;
+    }
+    /**
+     * Function for updating the status of the merging request.
+     */
+    private function update_status_and_log(int $idrecord,
+                                            int $status,
+                                            array $log): void {
+        global $DB;
+        $update = (object)[
+            'id' => $idrecord,
+            'status' => $status,
+            'log' => json_encode($log),
+        ];
+        if ($status == merge_request::COMPLETED_WITH_SUCCESS ||
+            $status == merge_request::COMPLETED_WITH_ERRORS) {
+                $update->timecompleted = time();
+        }
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            $update
+        );
+    }
+    /**
+     * Function for verifying the users to keep and remove.
+     */
+    private function verify_users_to_keep_and_remove(merge_request $mergerequest): void {
+        if (is_null($mergerequest->removeuserid)) {
+            $this->removeuserid = $this->find_user_id_or_fail($mergerequest->removeuserfield,
+                                                                $mergerequest->removeuservalue,
+                                                                $mergerequest->retries
+                                                                );
+        }
+        if (is_null($mergerequest->keepuserid)) {
+            $this->keepuserid = $this->find_user_id_or_fail($mergerequest->keepuserfield,
+                                                                $mergerequest->keepuservalue,
+                                                                $mergerequest->retries
+                                                            );
+        }
+    }
+    /**
+     * Function for updating number of retries.
+     */
+    private function update_retries(object $mergerequest): void {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $mergerequest->id,
+                'retries' => $mergerequest->retries + 1,
```

Comment:
the only place we call this method, we have already done the `retries + 1`. So, here we have to have simply `'retries' => $mergerequest->retries,`